### DISCUSSION TITLE
rust-course/usecases/aws-rust

### DISCUSSION THREAD

**Discussion by sunface** [sunface/rust-course#637] [章节评论区]
https://course.rs/usecases/aws-rust.html

**Comment by Gavin0x0** [comment]
前排

**Comment by mxismean** [comment]
学习

**Comment by OliverChao** [comment]
> Rust 在去年下半年被纳入了 Linux 的官方开发语言

貌似并没有, 而且在2022年,rust也没有被纳入内核官方开发语言.
在网上查到的资料只是说. linus对rust比较看好, 而且对采用rust持开放态度, 还说会采用其他语言编写一些其他的模块,但不一定是rust.
以linus的怪脾气, 估计能打动他的只有代码了, 就看2022年rust能不能拿出来了

**Reply by sunface** [reply]
已经纳入了～只不过现在并没有达到 C 的地位罢了

**Reply by keijack** [reply]
已经纳入了，但是毕竟有那么多 C 代码，Rust 不可能一时能达到 C 的地位的。

**Reply by dejunqi2008** [reply]
https://www.zdnet.com/article/linus-torvalds-rust-will-go-into-linux-6-1/

**Comment by unlimitedcodeG** [comment]
veriy nice.

**Comment by shazi4399** [comment]
学习

**Comment by 1793523411** [comment]
环保与可持续发展的rust

**Comment by WanhaoZhang** [comment]
那rust有什么缺点吗？或许可以讨论一下。

**Reply by matyle** [reply]
难写兄弟

**Reply by soloist-v** [reply]
写起来还可以，想要写起来没障碍，就得用到unsafe，如果不用那就得熟悉rust提供的黑魔法，两个是等价的，黑魔法基本也是unsafe实现的，但是可能要比你实现的安全。

**Reply by kinstom1232** [reply]
编译太慢了

**Reply by araraloren** [reply]
上班的时候编译太快了，给资本家节省时间了

**Comment by ttxs25830** [comment]
等一下，那张表太奇怪了，typescript执行时间比js慢了7倍？我完全不能理解！这是把代码从ts编译成js的时间也算进去了？

**Reply by framist** [reply]
+1

**Reply by re0312** [reply]
看了一下论文原文，js和ts在大部分基准测试中差不多，在fannkuch-redux中ts比js慢了一个数量级拉低了总评分，但是理论上fannkuch-redux这个测试ts和js也不会有太大差距，估计是测试的时候实现上或者使用的运行时有差异。

**Comment by soloist-v** [comment]
c++离开作用域总是会执行析构函数，但是rust的move机制让析构函数只执行1次，rust的宏+泛型比c++的模板强大很多，也把跟开发无关的代码隐藏了，专注做该做的事情而不是老在那看template代码，没有意义，还没有rust的宏强大，虽然c++也有std::move但它是运行时的需要配合右值引用构造和右值引用赋值运算符仍然有运行代价。个人觉得c++应该得到rust的编译期move和rust的宏，同时摒弃template改用泛型和宏，还有类似cargo的的东西，这样才能符合它的“0成本抽象”的理念。

**Comment by Bruce012356** [comment]
rust是语言基本的finops！

**Reply by Bruce012356** [reply]
rust是语言级别的finops！

**Comment by wangjunstf** [comment]
good job

**Reply by goranguom** [reply]
如何说服公司使用rust ？

**Comment by oswaldoooo** [comment]
rust上手了一段时间感觉还是比较ok的，除了lifetime，lifetime那里有点迷，其它很多都可以从c++和go的视角代入来看，rust的宏简直就是我的神，我go使用的是我自己编写的“宏”来生成go代码的，go的运行时反射我也在用，但还是不仅人意，很多地方明明能在编译期间做的，它就只能在运行时用反射，所以我后来自己搞了个"宏"生成代码，但是新问题又来了，生成的代码必须实际写入go文件，又会造成代码冗杂，后面的其他同事接手代码又会懵逼。而rust的宏是比较合理的，用syn将tokenstream解析为ast语法树，然后用quote将需要生成的代码走模版返回而不是跟胶水一样东瓶西凑

**Reply by liigo** [reply]
虽然lifetime在C/C++语言里不是一个显式存在的概念，但是好的C/C++程序员的脑子里一定会有lifetime的概念，否则他是用不好指针的。Rust编译器试图在编译期理解lifetime，在某些特定场景下的确有些问题，给程序员造成困扰，但也在逐步改进。

**Comment by light-glory** [comment]
rust已经正式加入linus内核6.14的开发了
